velox/vector/BaseVector.h

```diff
@@ -863,6 +863,18 @@ class BaseVector {
     storageByteCount_ = std::nullopt;
   }
 
+  // Slice a buffer with specific type.
+  //
+  // For boolean type and if the offset is not multiple of 8, return a shifted
+  // copy; otherwise return a BufferView into the original buffer (with shared
+  // ownership of original buffer).
+  static BufferPtr sliceBuffer(
```

Comment:
Are you making this API public? If so, please, extract this change into a separate PR and add tests.

Reply:
Create PR for this: https://github.com/facebookincubator/velox/pull/11001, please help to take a look, thanks.